### DISCUSSION TITLE
Module: Base64 Decoder

### DISCUSSION THREAD

**Discussion by TheTechromancer** [blacklanternsecurity/bbot#729] [Ideas]
This module would detect base64, decode it, and re-raise it in a format that can be decoded/excavated again.

In order for this to work properly, we'd need to implement a generic text data-type, similar to `RAW_RIR_DATA` in Spiderfoot. Once this was implemented, BBOT's recursion would effectively create a feather-duster-like functionality where multi-layered encodings could be recurisvely decoded and juicy secrets extracted from them.

This could be implemented as either an excavate submodule, or its own standalone (internal?) module. Supporting other encoding types besides base64, such as URL-encoding, could also be considered. It could also support extracting compressed data such as gzip, etc.

* [Spiderfoot `base64` module](https://github.com/smicallef/spiderfoot/blob/master/modules/sfp_base64.py)
* [Spiderfoot `binstring` module](https://github.com/smicallef/spiderfoot/blob/master/modules/sfp_binstring.py)

**Comment by liquidsec** [comment]
I think envelopes fulfilled this nicely, going to go ahead and close
